index.js

```diff
@@ -85,20 +85,112 @@ module.exports = {
           .catch(this._errorMessage.bind(this));
       },
 
-      activate(/* context */) {
+      activate(context) {
         const revisionKey = this.readConfig('revisionKey');
 
         this.log(`preparing to activate ${revisionKey}`, {
           verbose: true
         });
 
-        return this._uploadDeployInfo(this.s3)
-          .then(() => {
-            this.log(`✔  activated revison ${revisionKey}`, {
-              verbose: true
-            });
-          })
-          .catch(this._errorMessage.bind(this));
+        let _this = this;
+        return this.fetchRevisions(context).then(function(revisions) {
+           let found = revisions.revisions.map(function(element) { return element.revision; }).indexOf(revisionKey);
+           if (found >= 0) {
+
+             return _this._uploadDeployInfo(_this.s3)
+               .then(() => {
+                 _this.log(`✔  activated revison ${revisionKey}`, {
+                   verbose: true
+                 });
+               })
+               .catch(_this._errorMessage.bind(_this));
+
+           } else {
+             return RSVP.reject("REVISION NOT FOUND!"); // see how we should handle a pipeline failure
+           }
+         });
+      },
+
+      fetchRevisions(context) {
+        return this._list(context)
+          .then(function(revisions) {
+            return {
+              revisions: revisions
+            };
+          });
+      },
+
+      _list(/* context */) {
+        const bucket = this.readConfig('bucket');
+        const deployArchive = this.readConfig('deployArchive');
+        const archiveExt = `.${this.readConfig('archiveType')}`;
+        const deployInfo = this.readConfig('deployInfo');
+        const prefix = this.readConfig('prefix');
+        const archivePath = prefix ? [prefix, deployArchive].join('/') : deployArchive;
+        const indexKey = prefix ? [prefix, deployInfo].join('/') : deployInfo;
```

Comment:
Thanks for pointing out the `prefix`, it also showed me that I was missing `deployInfo`.  In our deploy config we have `deployArchive` and `deployInfo` setup to the same prefix so I missed the default `dist` scenario.

Reply:
Ah okay cool, glad to hear this helped!